### ISSUE TITLE
Is it possible to refactor a filename i.e. move files around while project-wide imports get updated?

### ISSUE THREAD

**Issue by NANASHI0X74** [syl20bnr/spacemacs#14118]
I was just wondering this, but I can't find a keybinding for it under SPC m in dart mode using lsp backend and when I tried moving a file via treemacs, not only do the imports not get updated, the dart analyzer seems to think it's still there... Even though I'm usually shown an error when an import target doesn't exist, I don't get errors for imports from a file I moved somewhere else

**Comment by NANASHI0X74** [comment]
I also just realized that when I move stuff in treemacs and have that file open in a buffer, that buffer's underlying file path doesn't get updated, so emacs assumes the buffer has unsaved changes, which is kinda annoying as well

**Comment by lebensterben** [comment]
SPC f R?

**Comment by NANASHI0X74** [comment]
that solves the issue with the buffer's file path but unfortunately doesn't update import directives :/

**Comment by NANASHI0X74** [comment]
looks like this this is a feature that some lsp-based language modes support and others don't. For dart, they don't yet and it looks like they're blocked by another issue on lsp itself. So, closing.

**Comment by NANASHI0X74** [comment]
ok the PR I was talking about is merged now:
https://github.com/microsoft/language-server-protocol/pull/1074
but I suppose it'll take some time until that's incorporated into lsp-mode and dart-analyzer/dart-mode
